hikari/impl/event_manager_base.py

```diff
@@ -260,19 +260,24 @@ async def _invoke_callback(
             trio = type(ex), ex, ex.__traceback__.tb_next if ex.__traceback__ is not None else None
 
             if base_events.is_no_recursive_throw_event(event):
-                _LOGGER.error("an exception occurred handling an event, but it has been ignored", exc_info=trio)
+                _LOGGER.error(
+                    "an exception occurred handling an event (%s), but it has been ignored",
+                    type(event).__name__,
+                    exc_info=trio,
+                )
             else:
-                _LOGGER.error("an exception occurred handling an event", exc_info=trio)
-                await self.dispatch(
-                    base_events.ExceptionEvent(
-                        app=self._app,
-                        shard=getattr(event, "shard") if isinstance(event, shard_events.ShardEvent) else None,
-                        exception=ex,
-                        failed_event=event,
-                        failed_callback=callback,
-                    )
+                exception_event = base_events.ExceptionEvent(
+                    app=self._app,
+                    shard=getattr(event, "shard") if isinstance(event, shard_events.ShardEvent) else None,
```

Comment:
do we need to use getattr here?

Reply:
yes, because there is no guarantee the event has that attribute in the first place, so it would cause MyPy to complain.

Reply:
why would it cause mypy to complain when you've lowered the scope of the variable with an instance check?